solr/core/src/test/org/apache/solr/response/transform/TestChildDocTransformer.java

```diff
@@ -81,8 +81,8 @@ private void testChildDoctransformerXML() {
 
     String test3[] = new String[] {
         "//*[@numFound='1']",
-        "/response/result/doc[1]/doc[1]/str[@name='id']='3'" ,
-        "/response/result/doc[1]/doc[2]/str[@name='id']='5'" };
+        "/response/result/doc[1]/doc[1]/str[@name='id']='5'" ,
```

Comment:
This had to be changed since we initialise the for loop in ChildDocTransformer using the limit param, so the sort is docID descending now.
Thought it is important to point it out, and perhaps this should also be documented.

Reply:
I suspect you are simply using the "sort" word inappropriately.  "sort" is related to ordering, but we didn't change the order.  We did change the "window" or "offset", if you will, into the docs to examine.  Ordering/sort hasn't changed.

Reply:
Yes,
I meant the order in which the documents are counted.
My bad.

Reply:
Again, the order hasn't changed ;-)

Reply:
since we initialize the document using from the highest index to exhaust the limit if possible, it means some documents may be skipped. If we have 3 child docs: 1 , 2, 3, and the limit is set to "2", only doc 2 and 3 will be added if we initialise the index to the one that will exhaust the limit. Another way is simply to count the number of matching docs so far, and continue onto the next root doc if reached(return from the transform method).

Reply:
I guess I was not specific enough beforehand.

Reply:
"If we have 3 child docs" ....   -- the explanation in that sentence is perfectly clear, I think.  The next sentence confused me, but whatever.

Reply:
The next sentence is a proposal for a different way to limit the number of child documents.
Incase the current logic is sufficient, then we have no need to worry about it.